cmake/CompileBoost.cmake

```diff
@@ -1,26 +1,29 @@
-find_package(Boost 1.72)
+if(Boost_ROOT)
+  message(STATUS "Using Boost at location: ${Boost_ROOT}")
+  set (BOOST_INCLUDE_DIR ${Boost_ROOT})
+  include_directories(${Boost_ROOT})
```

Comment:
Setting `include_directories` globally is not good style - it means that any component we built will be able to use boost

Reply:
Newer versions of CMakes encourage using `target_include_directories`, or in general, target-specific properties.

https://cmake.org/cmake/help/latest/command/target_include_directories.html